sharding/nodesCoordinator/indexHashedNodesCoordinator.go

```diff
@@ -1037,11 +1047,21 @@ func (ihnc *indexHashedNodesCoordinator) computeShardForSelfPublicKey(nodesConfi
 	return selfShard, false
 }
 
-// ConsensusGroupSize returns the consensus group size for a specific shard
-func (ihnc *indexHashedNodesCoordinator) ConsensusGroupSize(
+// ConsensusGroupSizeForShardAndEpoch returns the consensus group size for a specific shard in a given epoch
+func (ihnc *indexHashedNodesCoordinator) ConsensusGroupSizeForShardAndEpoch(
 	shardID uint32,
+	epoch uint32,
 ) int {
-	currentChainParameters := ihnc.chainParametersHandler.CurrentChainParameters()
+	ihnc.mutNodesConfig.RLock()
```

Comment:
why is this locking required?
If it is only for the log ihnc.currentEpoch, I would change the log and remove the current epoch and remove the locking as well. I think more relevant for the warn would be to print the epoch for which the consensusGroupSize failed on.

Reply:
right, good catch. fixed